src/Response/BodyResponse.php

```diff
@@ -127,6 +128,8 @@ public function json(int $depth = 512, int $options = 0): Promise
 			} catch (JsonDecodeException $exception) {
 				$newDeferred->reject($exception);
 			}
+		})->catch(function (Throwable $reason) use ($newDeferred) {
```

Comment:
![image](https://user-images.githubusercontent.com/109361613/231564094-38a3966a-2a18-48b4-a393-32737ac22e73.png)
I added catch function at the end here in the srch/Response/BodyResponse.php